src/validator/LookupValidator.h

```diff
@@ -44,7 +44,10 @@ class LookupValidator final : public Validator {
 
     Status rewriteRelExpr(RelationalExpression* expr);
 
-    StatusOr<Value> checkConstExpr(Expression* expr, const std::string& prop);
+    StatusOr<Value> checkConstExpr(Expression* expr,
+                                   const std::string& prop,
+                                   const Expression::Kind kind,
+                                   const bool leftIsAE);
```

Comment:
Please think about it. Is it necessary to change it here?

Reply:
To determine whether a value should be cast, it is necessary to get the info about the type of relational expression and also the position of the label attribute.

Reply:
> To determine whether a value should be cast, it is necessary to get the info about the type of relational expression and also the position of the label attribute.

I think auto `type = schema->getFieldType(prop); `is enough. WDYT?

Reply:
```schema->getFieldType(prop)``` returns the data type of the column while the parameter```kind```is the kind of relational expression(GT, LT). In this function, ```Expression* expr``` is always a const expression, and I couldn't see another way to determine the comparison operator. Please correct me if I am wrong.